File: ObjcCrashCourse/InitStuff/InitStuff/Person.m
//
//  Person.m
//  InitStuff
//
//  Created by Allen Czerwinski on 5/1/16.
//  Copyright © 2016 Allen Czerwinski. All rights reserved.
//

#import "Person.h"

@implementation Person
-(id)initWithFirstName:(NSString*)first lastName:(NSString*)last {
    
    self.firstName = first;
    self.lastName = last;
    // We could put the super init here
//    self = [super init];
//    return self;
    
    // or we could create another initializer below and return it here
    return [self initWithAge:500];

}

-(id)initWithAge:(NSInteger)age {
    
    self = [super init];
    return self;
}

-(void)printName {
    NSLog(@"%@ %@", self.firstName, self.lastName);
}

@end

File: Swift2iOS9Udemy/NSUser Defaults Lecture/NSUser Defaults Lecture/Person.swift
//
//  Person.swift
//  NSUser Defaults Lecture
//
//  Created by Allen Czerwinski on 2/16/16.
//  Copyright © 2016 Allen Czerwinski. All rights reserved.
//

import Foundation

class Person: NSObject, NSCoding {
    var firstName: String!
    var lastName: String!
    
    init(first:String, last:String) {
        firstName = first
        lastName = last
    }
    
    override init() {
        
    }
    
    required convenience init?(coder aDecoder: NSCoder) {
        self.init()
        self.firstName = aDecoder.decodeObjectForKey("firstName") as? String
        self.lastName = aDecoder.decodeObjectForKey("lastName") as? String
    }
    
    func encodeWithCoder(aCoder: NSCoder) {
        aCoder.encodeObject(self.firstName, forKey: "firstName")
        aCoder.encodeObject(self.lastName, forKey: "lastName")
    }
    
}